k9mail/src/main/java/com/fsck/k9/mailstore/LocalFolder.java

```diff
@@ -998,6 +998,42 @@ public LocalMessage doDbWork(final SQLiteDatabase db) throws WrappedException, U
         }
     }
 
+    public Set<Long> getDeletedMessageUids() throws MessagingException {
```

Comment:
This is very similar to the method above. It should be possible to use one underlying method for most of this and just passing in a different query.

Reply:
I'm also grumpy that there's no test, but there's no test for any LocalFolder stuff. Maybe I should look at that.

Reply:
@philipwhiuk done !